### PR TITLE
correctly presenting progressbar when buffer scrolls and clearing at end

### PR DESCRIPTION
**Pull request by SteveL-MSFT**
Windows progressbar requires ability to cache the screen buffer contents so it can write it back once progress is done.  Linux doesn't provide a way for console apps to read the screen buffer (security concern).  Changes:
- scroll buffer appropriately by the size needed for progress bar if near bottom of screen, adjust progressbar location to account for scrolling
- create an empty screen buffer used to clear the progress bar once done

No easy way to add tests for ProgressBar.

addresses #1625 

Before the change:
before

After the change:
after

### CODE REVIEW

File: src/Microsoft.PowerShell.ConsoleHost/host/msh/ProgressPane.cs [hunk 1]
```diff
@@ -91,32 +91,43 @@ class ProgressPane
...
-                // replace the saved region in the screen buffer with our progress display
-                _location = _rawui.CursorPosition;
+                _location.Y = _rawui.CursorPosition.Y;
 
-                //set the cursor position back to the beginning of the region to overwrite write-progress
-                //if the cursor is at the bottom, back it up to overwrite the previous write progress
-                if (_location.Y >= _rawui.BufferSize.Height - rows)
+                //if the cursor is at the bottom, create screen buffer space by scrolling
+                int scrollRows = rows - ((_rawui.BufferSize.Height - 1) - _location.Y);
+                for (int i = 0; i < rows; i++)
```
Comment (lzybkr) [inline src/Microsoft.PowerShell.ConsoleHost/host/msh/ProgressPane.cs]: In rare cases, there may be junk on the screen below the prompt. Maybe we should be writing out spaces to make sure it's clear.

Comment (SteveL-MSFT) [inline src/Microsoft.PowerShell.ConsoleHost/host/msh/ProgressPane.cs]: If there's junk, it'll get overwritten anyways with `tempProgressRegion` where every cell has a character including spaces

**Review by lzybkr — APPROVED** [review]

**Comment by iSazonov** [comment]
> No easy way to add tests for ProgressBar.

We can try our TestHostCS.psm1

**Comment by SteveL-MSFT** [comment]
@iSazonov don't think TestHostCS.psm1 will help here since we're trying to validate how powershell.exe handles progressbar

**Comment by lzybkr** [comment]
Testing this would be similar to how I test PSReadline - I introduce a mock console interface, so instead of `Console.Write`, you write something more like `_console.Write` - then the test framework has a mock for the interface (which doesn't really exist, you need to invent one.)

It's a pain to write these tests and there is little return in that investment, so that's why I didn't ask Steve for any tests.

**Comment by iSazonov** [comment]
>there is little return in that investment

If the stock price change, I may try to enhance our  TestHostCS.psm1 to support this scenarios. 😄

**Pull request merged**